rbac/rbac/dev_middleware.py

```diff
@@ -41,8 +41,8 @@ def process_request(self, request):  # pylint: disable=no-self-use
                 "identity": {
                     "account_number": "10001",
                     "org_id": "11111",
-                    "type": "User",
-                    "user": {
+                    "type": "Associate",
```

Comment:
Ah! I see that _all_ endpoints are currently working with it set this way (even external), because of the change in the middleware to conditionally set `user_info`. 

The problem is that the internal identity is not the same shape. It will actually look something like:
```json
{
  "identity": {
    "associate": {
      "Role": [
        "role"
      ],
      "email": "associate_dev@foo.com"
    },
    "auth_type": "saml-auth",
    "type": "Associate"
  }
}
```

On that note, we should probably have a way to flip between an external identity and internal identity, not just flipping the `associate` key/value. Otherwise I think we'll run into these types of issues.

Reply:
I actually worked up a way to do that some time ago, but let it go by the wayside. I think I triggered it on a header for what kind of identity we wanted to use in dev, does that seem reasonable? If so, I'll get it recoded and push it up